pkg/cmd/waitforceo/waitforceo.go

```diff
@@ -55,8 +55,5 @@ func (w *waitForCeoOpts) Run() error {
 	if err := bootstrapteardown.WaitForEtcdBootstrap(context.TODO(), config); err != nil {
 		klog.Errorf("error waiting for cluster-etcd-operator %#v", err)
 	}
-	if err := bootstrapteardown.WaitForKubeApiServerRollout(context.TODO(), config); err != nil {
```

Comment:
Explain why this isn't needed in this case.  Also, explain where an why this is used.

Reply:
https://github.com/openshift/installer/blob/993c8cec1e8b1dbf58383fa52b264695187656be/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L403

It was one of the design decisions that to be safe, we will scale the operator from bootstrap node to all the three masters (4 node etcd cluster) and only then remove the bootstrap node. We need to make the installer wait for this scaling, if not it was destroying the bootstrap vm while CEO is still scaling.


I moved the functionality of waiting for Kube API server into the teardown controller. The teardown controller will only report `BootstrapRemoved` [1] condition when it has waited for etcd scaling[2] and Kube API server[3] rollout with the valid config.

1. https://github.com/openshift/cluster-etcd-operator/pull/63/files#diff-269001f9edf77680f8464a9c8a12d252R149
2. https://github.com/openshift/cluster-etcd-operator/pull/63/files#diff-269001f9edf77680f8464a9c8a12d252R118
3. https://github.com/openshift/cluster-etcd-operator/pull/63/files#diff-269001f9edf77680f8464a9c8a12d252R133